File: java/lsp-01-refactored/src/main/java/no/bouvet/solid/lsp/BirdService.java
package no.bouvet.solid.lsp;

import java.util.List;

public class BirdService {

    public void feedThemAll(List<Bird> birds) {
        birds.forEach(bird -> bird.eat());
    }

    public void letThemGo(List<FlyingBird> flyingBirds) {
        flyingBirds.forEach(bird -> bird.fly());
    }
}

File: dotnet/Solid/Solid/LSP/Example1/BirdService.cs
using System.Collections.Generic;

namespace Solid.LSP.Example1
{
    public class BirdService
    {
        public void FeedThemAll(List<Bird> birds)
        {
            birds.ForEach(bird => bird.Eat());
        }

        public void LetThemGo(List<Bird> birds)
        {
            foreach (Bird bird in birds)
            {
                bird.Fly();
            }
        }
    }
}
